plugins/SegmentEditor/API.php

```diff
@@ -395,11 +395,37 @@ public function getAll($idSite = false)
             }
         }
 
+        $segments = $this->filterSegmentsWithDisabledElements($segments, $idSite);
         $segments = $this->sortSegmentsCreatedByUserFirst($segments);
 
         return $segments;
     }
 
+    /**
+     * Filter out any segments which cannot be initialized due to disable plugins or features
+     *
+     * @param array $segments
+     * @param       $idSite
+     *
+     * @return array
+     */
+    private function filterSegmentsWithDisabledElements(array $segments, $idSite = false): array
+    {
+        $sites = [];
+        if (!empty($idSite)) {
+            $sites = [$idSite];
+        }
+
+        foreach ($segments as $k => $segment) {
+            try {
+                new Segment($segment['definition'], $sites);
```

Comment:
I guess without having any structured way of recording what segment uses what dimensions/metrics and then using that when the providers of those are disabled this might be the best way how to test if a segment can be created.

Or did you consider any other approaches?

Reply:
The segment instantiation is where the problem exception is thrown and appears to be quite lightweight as it checks a list of cached segments to see if the segment is available. I could have replicated this logic in a slightly more efficient manner but then these two areas would need to be kept in sync. Without implementing some sort of segment dependency system this seems like the simplest and lowest risk approach.